apps/dashboard/src/components/workflow-editor/in-app-preview.tsx

```diff
@@ -38,44 +38,62 @@ export const InAppPreview = (props: InAppPreviewProps) => {
           <InboxSettings />
         </div>
       </div>
-      <div className="z-20 mb-2 p-2">
-        <div className="mb-2 flex items-center gap-2">
-          {data?.result?.preview && 'avatar' in data.result.preview && (
-            <img src={data.result.preview.avatar as string} alt="avatar" className="h-5 w-5 rounded-full" />
-          )}
-          {data?.result?.preview && 'subject' in data.result.preview ? (
-            <span className="text-xs font-medium text-neutral-600">
-              {(data.result.preview as InAppRenderOutput).subject}
-            </span>
-          ) : (
-            data?.result?.preview && 'body' in data.result.preview && <Body text={data.result.preview.body} />
-          )}
-        </div>
-        {data?.result?.preview && 'body' in data.result.preview && 'subject' in data.result.preview && (
-          <Body text={data.result.preview.body} />
-        )}
+      {data.result?.type === ChannelTypeEnum.IN_APP && (
+        <div className="z-20 mb-2 p-2">
+          <div className="mb-2 flex items-center gap-2">
+            {data.result.preview.avatar && (
+              <img src={data.result.preview.avatar as string} alt="avatar" className="h-5 w-5 rounded-full" />
+            )}
+            {data.result.preview.subject ? (
+              <Subject text={data.result.preview.subject as string} />
+            ) : (
+              <Body text={data.result.preview.body} />
+            )}
+          </div>
+
+          {data.result.preview.subject && <Body text={data.result.preview.body} />}
 
-        <div className="mt-3 flex items-center justify-end gap-1">
-          {data?.result?.preview && 'primaryAction' in data.result.preview && (
-            <Button className="text-xs font-medium shadow-none" type="button" variant="primary">
-              {(data.result.preview as InAppRenderOutput).primaryAction?.label}
-            </Button>
-          )}
-          {data?.result?.preview && 'secondaryAction' in data.result.preview && (
-            <Button variant="outline" className="text-xs font-medium" type="button">
-              {(data.result.preview as InAppRenderOutput).secondaryAction?.label}
-            </Button>
-          )}
+          <div className="mt-3 flex items-center justify-end gap-1">
+            {data.result.preview.primaryAction && (
+              <Button className="text-xs font-medium shadow-none" type="button" variant="primary">
+                {(data.result.preview as InAppRenderOutput).primaryAction?.label}
+              </Button>
+            )}
+            {data.result.preview.secondaryAction && (
+              <Button variant="outline" className="text-xs font-medium" type="button">
+                {(data.result.preview as InAppRenderOutput).secondaryAction?.label}
+              </Button>
+            )}
+          </div>
         </div>
-      </div>
+      )}
+    </div>
+  );
+};
+
+const parseText = (inputText: string) => {
```

Comment:
@desiprisg If I recall correctly we export a utility for this from @novu/js/internals. Let's reuse that.